### ISSUE TITLE
Holding up/down arrow to change pitch, glitches with larger scores

### ISSUE THREAD

**Issue by MqcCheeze** [musescore/MuseScore#22639]
### Issue type

General playback bug

### Bug description

(wow I'm creating a lot of bug reports, sorry devs lol)

So when I'm working on my score, I may need to change pitch by using the up and down arrows, sometimes instead of doing interval shortcuts with control + 4 (for example), I just hold the up arrow key. This creates a weird playback bug of the note where it keeps repeatedly playing the note for a short while, then finally correcting the sound after I stop holding the key. See video for reference since I can't describe well...

I don't have any issues with smaller scores.

### Steps to reproduce

1. Need a large score
2. Enter a note (any note is fine)
3. Hold up/down arrow to change the pitch.

### Screenshots/Screen recordings

Example:

https://github.com/musescore/MuseScore/assets/87596784/da4d0e80-76c8-447a-b7c9-a0587cd3f0c3

### MuseScore Version

4.2

### Regression

I don't know

### Operating system

Windows 11

**Comment by zacjansheski** [comment]
Also in this issue... could you please confirm if this is Muse Sounds only or both Muse Sounds and MS Basic?

Attaching an example score would also be helpful for reproducing.

**Comment by bkunda** [comment]
I think this should be re-tested again.

Interesting to compare current behaviour in master (First example is Muse Sounds; Second is MS Basic):

https://github.com/musescore/MuseScore/assets/86290556/a1cb2789-b844-4c6b-9154-a26cfa439fc2

...with the behaviour after the new note input optimisations (See https://github.com/musescore/MuseScore/pull/22791):

https://github.com/musescore/MuseScore/assets/86290556/b5f62273-28fa-4383-a6bb-495809e2ded7

In short, I cannot reproduce this problem anymore (at least on my machine running macOS). 

@DmitryArefiev would be good if you could pls take a look at reproducing this on a lower-spec machine so we can see if it's actually resolved by @RomanPudashkin and @konradglas's recent optimisation work.
